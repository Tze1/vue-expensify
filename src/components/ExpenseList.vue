<template>
  <div class="expenselist component">
    <div v-if="userLoggedIn">
      <p v-if="expenses.length === 0" class="expenselist-empty">[no expenses found]</p>
      <Expense v-else v-for="expense in expenses" :key="expense.id" :expense="expense"/>
    </div>
    <div v-else class="expenselist-public">
      <p>PRIVATE CONTENT!  Please <router-link to="/">Log in</router-link> first.</p>
    </div>
  </div>
</template>

<script>
  import { mapState } from 'vuex';
  import Expense from './Expense';

  export default {
    name: 'ExpenseList',
    computed: mapState({
      userLoggedIn () {
        return this.$store.getters.userLoggedIn;
      },
      expenses: state => state.expenses,
    }),
    components: {
      Expense
    },
  };
</script>

<style scoped lang="scss">
  @import '../styles/app.scss';
</style>
