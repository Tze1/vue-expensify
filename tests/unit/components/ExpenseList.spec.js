import { shallowMount } from '@vue/test-utils';
import testExpenses from '../../fixtures/testExpenses';
import ExpenseList from '@/components/ExpenseList';

const $store = {
  state: {
    auth: {
      user: {
        displayName: 'Test',
        photoUrl: '/img/avatar-default.png',
      },
    },
    expenses: testExpenses,
  },
  getters: {
    userLoggedIn: state => {
      return (state.auth.user !== undefined);
    },
    filteredExpenses: state => {
      return state.expenses;
    }
  },
};
const wrapper = shallowMount(ExpenseList, {
  mocks: {
    $store,
  },
});

describe('ExpenseList.vue', () => {
  it('renders ExpenseList component properly', () => {
    expect(wrapper).toMatchSnapshot();
  })
});
