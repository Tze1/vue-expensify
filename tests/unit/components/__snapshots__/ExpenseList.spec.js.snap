// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseList.vue renders ExpenseList component properly 1`] = `
<div class="expenselist component">
  <div></div>
</div>
`;
